lib/vf_analyzers.cpp

```diff
@@ -527,6 +527,8 @@ struct ValueFlowAnalyzer : Analyzer {
                 continue;
             if (exact && v.intvalue != 0 && !isPoint)
                 continue;
+            if (astIsUnsigned(tok) != astIsUnsigned(v.tokvalue))
```

Comment:
This is fine, it is a little too restrictive which could lead to FNs, but it will fix FPs. To fix FNs, we would need to check if the value is within the range of an unsigned value. I am not sure how to write a test case to demonstrate it though as most known values will be passed through with the afterAssign passes anyways. Need to think about it some more.

Reply:
+1 We could also wait until after the next release, so we can see the effects in daca.